pkg/storage/utils.go

```diff
@@ -0,0 +1,20 @@
+package storage
+
+import (
+	"strings"
+)
+
+func matchesNameOrID(filter, ID string, names []string) bool {
+	if filter == ID {
+		return true
+	}
+	for _, name := range names {
+		if filter == "" || filter == name {
```

Comment:
Why add a condition `filter == ""` here? I mean you have checked filter is not empty string in `ListImage`, and if we really need to check it here, why not together with `filter == ID`, IIRC `filter == ""` means no filter at all, maybe don't need to check it for every name.